content/post/asdf.md

```diff
@@ -0,0 +1,240 @@
+---
+author: vadim
+date: 2018-11-10
+title: "Advanced Scientific Data Format"
+image: /post/asdf/logo.png
+description: "What is ASDF, why it is awesome, why you should and how to use it. ASDF in source{d} projects."
+categories: ["technical"]
+---
+
+There is a project we are developing at source{d} named [Modelforge](https://github.com/src-d/modelforge).
+Its goal is to abstract the serialization and the retrieval of the machine learning
+models from the users. That is, it solves the following two major problems
+every ML engineer hits:
+
+1. What to save a trained model and load it back.
+2. How to reference and distribute trained models.
+
+Let's not go offtopic with discussing (2) - there are many funny subproblems there
+which deserve a dedicated blog post about Modelforge. Instead, the focus will be on
+serialization. The Python crowd is used to solve (1) with
+[`pickle`](https://docs.python.org/3/library/pickle.html) - the built-in
+serialization module. It works nicely while these three conditions hold:
+
+1. The origin of the files is trusted and secure.
+2. The interoperability with other languages is not needed.
+3. The serialized objects are not big.
+
+As some know very well, `pickle` contains a full-featured virtual machine inside,
+which is interpreted to recreate the serialized Python objects. The advantage
```

Comment:
the VM is interpreted? it's surprising (but maybe accurate)

Reply:
It is. So besides saving the object data you also save the instructions on how to materialize it in memory, e.g. type info.